data/extension/init/templates/tinygo/envoy.filters.http/default/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+var (
+	requestCounterName = "proxy_wasm_go.request_counter"
+	counter            proxywasm.MetricCounter
+)
+
+func main() {
+	proxywasm.SetNewRootContext(func(uint32) proxywasm.RootContext { return &context{} })
+	proxywasm.SetNewHttpContext(func(uint32) proxywasm.HttpContext { return &context{} })
+}
+
+type context struct{ proxywasm.DefaultContext }
+
+func (ctx *context) OnVMStart(int) bool {
+	var err error
+	counter, err = proxywasm.DefineCounterMetric(requestCounterName)
+	if err != nil {
+		proxywasm.LogCritical("failed to initialize request counter: ", err.Error())
+	}
+	return true
+}
+
+func (ctx *context) OnHttpRequestHeaders(int, bool) types.Action {
```

Comment:
Let's give parameters names (because that's the first question the users will have)